tools/releases/deprecate-release.yml

```diff
@@ -0,0 +1,68 @@
+trigger: none
+resources:
+  repositories:
+    - repository: GovernedTemplates
+      type: git
+      name: OneBranch.Pipelines/GovernedTemplates
+      ref: refs/heads/main
+  pipelines:
+    - pipeline: microsoft-teams-library-js-pipeline
+      source: 'microsoft-teams-library-js/M365 Platform/App SDK/OfficeDev.microsoft-teams-library-js'
+      project: ISS
+variables:
+  - name: System.Debug
+    value: true
+extends:
+  template: v2/Microsoft.Official.yml@GovernedTemplates
+  parameters:
+    platform:
+      name: m365
+      workload: Substrate
+      serviceTreeId: $(serviceTreeId)
+      serviceGroupName: Platform
+    stages:
+      - stage: Prod_Lockbox_Approval_Deployment
+        displayName: Lockbox Approval/Deployment
+        dependsOn: []
+        variables:
+          stage_type: deployment
+          azure_subscription_ids: $(subscriptionId)
+        jobs:
+          - job: Agent_job
+            pool:
+              type: release
+            steps:
+              - download: microsoft-teams-library-js-pipeline
+              - task: prepare-deployment@1
+                displayName:
+                inputs:
+                  taskType: credentialFetchTaskAzureRM
+                  armserviceconnection: $(serviceConnectionId)
+                  subscriptionid: $(subscriptionId)
+              - task: AssetRetention@3
+                displayName: ARtifact Retention Orchestrator Workflow (ARROW)
+                inputs:
+                  ArrowServiceConnection: $(arrowServiceConnectionId)
+                  IsShipped: true
+              - task: AzureKeyVault@2
+                displayName: 'Azure Key Vault: $(keyVaultName)'
+                inputs:
+                  ConnectedServiceName: $(serviceConnectionId)
+                  KeyVaultName: $(keyVaultName)
+                  SecretsFilter: $(keyVaultSecretsFilter)
+              - task: NodeTool@0
+                displayName: Use Node 18.x
+                inputs:
+                  versionSpec: 18.x
+              - task: PowerShell@2
+                displayName: Update npmrc with NPM-TOKEN
+                inputs:
+                  targetType: 'inline'
+                  script: Set-Content -Path $(PIPELINE.WORKSPACE)/microsoft-teams-library-js-pipeline/NPMFeed/.npmrc -Value "//registry.npmjs.org/:_authToken=$(NPM-TOKEN)"
+              - task: Npm@1
+                displayName: Publish to npm (tag beta) KV
+                inputs:
+                  command: custom
+                  workingDir: $(PIPELINE.WORKSPACE)/microsoft-teams-library-js-pipeline/NPMFeed
+                  verbose: false
+                  customCommand: deprecate @microsoft/teams-js@3.0.* "Package no longer supported. Use at your own risk"
```

Comment:
>deprecate @microsoft/teams-js@3.0.* [](http://example.com/codeflow?start=33&length=35)

Can you say more about how this pipeline works? This looks like every time the pipeline runs, it deprecates all 3.0.* packages. Is that how it's supposed to work? Or is the person who runs the pipeline supposed to be able to specify what version they want to deprecate?

Reply:
Sorry this was just meant to be a draft PR to see if the new pipeline would be added into ADO. Will be modifying this so that we can specify the version/versions we wish to deprecate as a Pipeline resource going forward.